### PR TITLE
Concurrent translation of multiple keys

### PR DESCRIPTION
**Pull request by hleft**
This is the simplest case of concurrent translation, no error handling, no resume function, can be used with care, if all goes well, it will be very fast
Or wait for https://github.com/yihong0618/bilingual_book_maker/pull/62

Usage: --max_procs num Note that first -h look at

If this parameter is not used, the program functions unchanged

If it is used, the output is modified because it is no longer outputting the content before and after translation because it is too fast.
Instead, it outputs the file inside the epub currently being processed and the apikey used for the current request (so you can see how fast it is now)

Note that it does not care how many keys you have, if you only have 3 keys, but concurrently 100, there will be problems
may be 30 keys, concurrently <= 3

It may output the wrong progress but the result book is correct, I think it should not output the progress bar, but it sometimes outputs... I don't know why.

chatgptapi_translator.py change a lot because https://github.com/yihong0618/bilingual_book_maker/pull/112, actually only add noprint option in this file

> My experience I use 30 keys --max_procs 5 to translate animal_farm.epub successfully every time. But when I translated a 300-page English book, the translation failed using --max_procs 5, so I changed it to --max_procs 3 and the translation was successful, but the translation time was only half of --max_procs 5.  --max_procs 3 is not three times as fast as --max_procs 1, but much faster

### CODE REVIEW

**Comment by yihong0618** [comment]
I will take a look at #112 first maybe tonight

**Comment by hleft** [comment]
speed test:

wc -w ：151581
wc -c：1201147
16.08s user 0.89s system 0% cpu 1:10:30.73 total
args: --max_procs 2

File: book_maker/translator/chatgptapi_translator.py [hunk 1]
```diff
@@ -14,49 +13,47 @@ def __init__(self, key, language, api_base=None):
 
     def rotate_key(self):
         openai.api_key = next(self.keys)
+        print(openai.api_key)
```
[the remark this answers is not in the document]
Comment (hleft) [reply book_maker/translator/chatgptapi_translator.py:16]: > don't print the key ....

Okay, this is to see the real speed of sending requests. I don’t know whether to add a number to show the number of requests

[the remark this answers is not in the document]
Comment (yihong0618) [reply book_maker/translator/chatgptapi_translator.py:16]: or replace the middle of the key to **************

**Review by Arborym — CHANGES_REQUESTED** [review]

**Comment by hleft** [comment]
A bit busy right now..

**Comment by hleft** [comment]
Now a lot of codes have changed, and I am using this function locally and no interest in improving this pr and fix conflicts. I feel that modifying it is not as good as raising a new pr, so I closed it and welcome others to pr

**Pull request closed without merging**